x-pack/test/functional/apps/discover/feature_controls/discover_security.ts

```diff
@@ -448,6 +449,19 @@ export default function (ctx: FtrProviderContext) {
         await globalNav.badgeExistsOrFail('Read only');
 
         // can't access logstash index directly
+        // swapping index patterns so we get an updated field list
+        // this is necessary since we don't have access to data view management
+        // nor can we force reload the browser in a test
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await PageObjects.discover.selectIndexPattern('logstash-*');
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await PageObjects.discover.selectIndexPattern('alias-logstash-discover');
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await PageObjects.discover.selectIndexPattern('logstash-*');
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await PageObjects.discover.selectIndexPattern('alias-logstash-discover');
+
+        await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
We should improve this, so it's not tied to data view switching behavior. This can be done by e.g. creating and using a different data view with a slightly different index pattern for this test. We are creating an alias index pattern here

https://github.com/elastic/kibana/blob/b8a23eb6c3dd60f88dced04af211a78dc70113fa/x-pack/test/functional/apps/discover/feature_controls/discover_security.ts#L392-L395

How about creating and using a data view with an index pattern `logstash*` instead of `logstash-*` in this test, because we wouldn't need to refresh the field list like this

Reply:
>How about creating and using a data view with an index pattern logstash* instead of logstash-* in this test, because we wouldn't need to refresh the field list like this

I think the thing that makes this difficult is also what makes the test valuable.